rclpy/rclpy/executors.py

```diff
@@ -205,7 +206,7 @@ def shutdown(self, timeout_sec: float = None) -> bool:
                 # Tell executor it's been shut down
                 self._guard.trigger()
 
-        if not self._work_tracker.wait(timeout_sec):
+        if not self._work_tracker.wait(timeout_sec, is_shutdown=self._is_shutdown):
```

Comment:
Maybe don't call wait if we're shutting down

Reply:
if we take this path, probably we do not need to call this method.